### PR TITLE
♻️ repo cleanup and doc

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

After reviewing the repo structure and the release workflow, I did the following changes:

- Reorganized and doc the repository to resemble osparc-simcore layout
- Started migration to  `pyproject.toml` (limited by dynamic setup of versioning)
- Added `.vscode` config templates
- Updated `Makefiles`

 
@Konohana0608  just FYI: this PR will still not bring the changes you need in the API.

## Related issue/s

- Preparations for ITISFoundation/osparc-issues#1611

### CODE REVIEW

File: .github/workflows/build-python-client.yml [hunk 1]
```diff
@@ -18,24 +18,27 @@ jobs:
...
-          echo "${version}" > clients/python/client/VERSION
-      - name: Generate client
+          echo "${version}" > clients/python/VERSION
+      - name: Auto-generate python_client and build distributions
         run: |
           source .venv/bin/activate
           cd clients/python
-          make install-dev
-          make dist-ci
+          uv pip install -r requirements/dev.txt
```
Comment (sanderegg) [inline .github/workflows/build-python-client.yml:39]: Question : is uv not used in the male target? That would make it more 1:1 with what you do as dev.
Also there is a GitHub action to install uv

File: .github/workflows/build-python-client.yml [hunk 2]
```diff
@@ -70,6 +73,8 @@ jobs:
         uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
+      - name: install uv
+        uses: yezz123/setup-uv@v4
```
Comment (sanderegg) [inline .github/workflows/build-python-client.yml:77]: Ah ok. But is that not installed after usage?

File: .github/workflows/build-python-client.yml [hunk 3]
```diff
@@ -90,6 +95,7 @@ jobs:
           python -m pip install clients/python/artifacts/dist/${{needs.build.outputs.osparc}} --find-links=clients/python/artifacts/dist
           cd clients/python
           make install-unit-test
+          pip list | grep osparc
```
Comment (sanderegg) [inline .github/workflows/build-python-client.yml:98]: Uv pip list?

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -27,16 +27,13 @@ repos:
...
-        args: [ --fix, --exit-non-zero-on-fix ]
-  - repo: https://github.com/pycqa/isort
-    rev: 5.12.0
-    hooks:
-      - id: isort
-        name: isort (python)
-        args: ["--profile", "black"]
+        args:
+          - --fix
+          - --exit-non-zero-on-fix
```
Comment (sanderegg) [inline .pre-commit-config.yaml:35]: We should add that in simcore too

File: .vscode/settings.template.json [hunk 5]
```diff
@@ -1,4 +1,15 @@
 {
+    "files.associations": {
+        ".*rc": "ini",
+        ".env*": "ini",
+        "**/requirements/*.in": "pip-requirements",
+        "**/requirements/*.txt": "pip-requirements",
+        "*logs.txt": "log",
+        "*.logs*": "log",
+        "*Makefile": "makefile",
+        "docker-compose*.yml": "dockercompose",
```
Comment (sanderegg) [inline .vscode/settings.template.json:10]: Nice!

File: clients/python/Makefile [hunk 6]
```diff
@@ -102,29 +95,41 @@ _check_venv_active:
 	# checking whether virtual environment was activated
 	@python3 -c "import sys; assert sys.base_prefix!=sys.prefix"
 
+
 .PHONY: install-dev
-install-dev: _check_venv_active ## install packages for development
-	pip install -r requirements/dev.txt
+install-dev: python-client _check_venv_active ## installs osparc_client and osparc in edit mode
+	python -m pip install -e artifacts/client
```
Comment (sanderegg) [inline clients/python/Makefile:101]: No uv here?

File: clients/python/Makefile [hunk 7]
```diff
@@ -102,29 +95,41 @@ _check_venv_active:
...
 
 .PHONY: pylint
 pylint: _check_venv_active ## runs linter (only to check errors. SEE .pylintrc enabled)
-	pylint --rcfile "$(PYTHON_DIR)/.pylintrc" -v "$(ARTIFACTS_DIR)/client"
+	pylint --rcfile "$(CLIENTS_PYTHON_DIR)/.pylintrc" -v src/osparc
 
 .PHONY: mypy
-mypy: $(SCRIPTS_DIR)/mypy.bash $(PYTHON_DIR)/mypy.ini ## runs mypy python static type-checker on this services's code. Use AFTER make install-*
-	@$(SCRIPTS_DIR)/mypy.bash client
+mypy: $(SCRIPTS_DIR)/mypy.bash $(CLIENTS_PYTHON_DIR)/mypy.ini ## runs mypy python static type-checker on this services's code. Use AFTER make install-*
```
Comment (sanderegg) [inline clients/python/Makefile:129]: Did you check the mypy.ini for the bad issue?

File: clients/python/pyproject.toml [hunk 8]
```diff
@@ -0,0 +1,37 @@
+[tool.mypy]
+# Global options
+check_untyped_defs = true
+disallow_any_generics = false
+# disallow_untyped_defs: if True, it enforces things like `def __init__(self) -> CLASSNAME` or `def test_() -> None` which does not worth the effort
+disallow_untyped_defs = false
+follow_imports = "silent"
+# ignore_missing_imports: removes all the missing imports stuff from external libraries which is annoying to the least
+ignore_missing_imports = true
```
Comment (sanderegg) [inline clients/python/pyproject.toml:9]: This one will bite you back heavily! That is precisely why we had such a mess in simcore

File: clients/python/pyproject.toml [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+warn_untyped_fields = true
+
+
+[tool.pytest.ini_options]
+# SEE https://docs.pytest.org/en/7.1.x/reference/customize.html#pyproject-toml
+testpaths = [
+    "test",
+]
+# SEE https://pytest-asyncio.readthedocs.io/en/stable/concepts.html#test-discovery-modes
+asyncio_mode = "auto"
```
Comment (sanderegg) [inline clients/python/pyproject.toml:36]: Ah you have it here. So I guess you can remove it from your pytest calls?

File: clients/python/setup.py [hunk 10]
```diff
@@ -1,41 +1,48 @@
...
-    f"osparc_client=={VERSION}",
     "httpx",
-    "tqdm>=4.48.0",
     "nest_asyncio",
-    "tenacity",
-    "pydantic>=2.0.0",
-    "pydantic-settings",
     "packaging",
+    "pydantic-settings",
+    "pydantic>=2.0.0",
```
Comment (sanderegg) [inline clients/python/setup.py:29]: Is this really necessary?

**Review by sanderegg — APPROVED** [review]
Be careful with ignoring imports with mypy. That is exactly how you get into many issues

**Review by matusdrobuliak66 — APPROVED** [review]
Very nice thank you! yes when we consolidate things now its easier to jump in and help :+1:

File: .github/workflows/build-python-client.yml (hunk 1 above)
Comment (pcrespov) [reply .github/workflows/build-python-client.yml:39]: @sanderegg what is the "male" target? :-D 
I did not have my PR udpated. I use `uv` everywhere where before was `pip`

File: clients/python/Makefile (hunk 7 above)
Comment (pcrespov) [reply clients/python/Makefile:129]: I am adding all the options in `pyproject.toml`

File: clients/python/pyproject.toml (hunk 9 above)
Comment (pcrespov) [reply clients/python/pyproject.toml:36]: yes

File: clients/python/setup.py (hunk 10 above)
Comment (pcrespov) [reply clients/python/setup.py:29]: no. i guess is from the old days.

**Pull request merged**